tests/test_format_notebook_content.py

```diff
@@ -20,6 +21,43 @@ def test_format_notebook_content():
     )
 
 
+def test_format_single_notebook():
+    """
+    Test formatting a single notebook in an directory of unformatted notebooks
+    formats only that notebook
+    """
+    data_path = pathlib.Path(f"{__file__}").parent / "data"
+    source_notebook_path = data_path / "unformatted" / "spaces.ipynb"
+    copy_notebook_path = data_path / "unformatted" / "spaces_copy.ipynb"
+    shutil.copy(str(source_notebook_path), str(copy_notebook_path))
+
+    source_json = json.loads(source_notebook_path.read_text())
+    copy_json = json.loads(copy_notebook_path.read_text())
+
+    assert all(
+        [
+            copy_cell["source"] == source_cell["source"]
+            for copy_cell, source_cell in zip(copy_json["cells"], source_json["cells"])
+        ]
+    )
+
+    # Have to run blackbook main to show spaces_copy.ipynb is unaffected
+    blackbook.__main__.main(source_notebook_path)
+    output_json = json.loads(source_notebook_path.read_text())
+    check_copy_json = json.loads(copy_notebook_path.read_text())
+    shutil.copy(str(copy_notebook_path), str(source_notebook_path))
+    pathlib.Path.unlink(copy_notebook_path)
```

Comment:
I understand that these two lines undo the changes to the unformatted notebook and delete the copy but it took me a while to figure that out 😅 I am not a huge fan of inline comments but maybe one is needed here.

I also suggest moving these two lines after the assert. That way the changes will be undone if the test passes 👍

Reply:
> I am not a huge fan of inline comments but maybe one is needed here.

Sure, can do.

> I also suggest moving these two lines after the assert. That way the changes will be undone if the test passes

I don't think you want that. If the test fails then the notebook is in an altered state and you'd have to reset the repo before you could run tests again. The tests should be as atomic as possible.